components/builder-api/src/server/resources/package_channels.rs

```diff
@@ -77,16 +78,24 @@ impl PackageChannels {
     }
 }
 
-fn promote_package(req: HttpRequest<AppState>) -> HttpResponse {
+fn promote_package(req: HttpRequest<AppState>) -> FutureResponse<HttpResponse> {
     let (origin, channel, pkg, version, release) =
         Path::<(String, String, String, String, String)>::extract(&req)
             .unwrap()
             .into_inner();
 
-    match do_promote_package(&req, origin, channel, pkg, version, release) {
-        Ok(_) => HttpResponse::new(StatusCode::OK),
-        Err(err) => err.into(),
-    }
+    let ident = format!("{}/{}/{}/{}", origin, pkg, version, release);
```

Comment:
might be a better way to handle this - if there is, I'd love to know about it

Reply:
```rust
let i = PkgIdent::new(origin, pkg, Some(version), Some(release));
let i_as_string = format!("{}", i);
```
Bonus poitns if `PromotePackage` could take a `PkgIdent` rather than a string.

Reply:
I think we could leverage PackageIdent for this